scripts/table_updates/update_data_table.py

```diff
@@ -37,42 +50,56 @@
 }
 
 
-def get_main_genie_clinical_sample_file(
+def get_main_genie_clinical_file(
     syn: synapseclient.Synapse,
     release: str,
     release_files_table_synid: str,
+    form: str,
+    column_mapping_table: pandas.DataFrame,
     logger: logging.Logger = None,
 ) -> pandas.DataFrame:
-    """This retrieves the main genie clinical sample file from consortium release
+    """This retrieves the main genie clinical patient or sample file from consortium release
 
     Args:
         syn (synapseclient.Synapse): synapse client connection
         release (str): release version to pull from for main genie
         release_files_table_synid (str): synapse id of the data relese files table
         from main genie
+        form (str): form name, can be either patient_characteristics or cancer_panel_test
+        column_mapping_table (pandas.DataFrame): GENIE BPC Elements Mapping table
         logger (logging.Logger): custom logger. Optional.
 
     Returns:
         pandas.DataFrame: the read in clinical file as dataframe
     """
-    release_files = download_synapse_table(syn, table_id=release_files_table_synid)
-    clinical_link_synid = release_files[
-        (release_files["release"] == release)
-        & (release_files["name"] == "data_clinical_sample.txt")
-    ]["fileSynId"].values[0]
+    release_files = utilities.download_synapse_table(syn, release_files_table_synid)
+    if form == "patient_characteristics":
+        clinical_link_synid = release_files[
+            (release_files["release"] == release)
+            & (release_files["name"] == "data_clinical_patient.txt")
+        ]["fileSynId"].values[0]
+    else:
+        clinical_link_synid = release_files[
+            (release_files["release"] == release)
+            & (release_files["name"] == "data_clinical_sample.txt")
+        ]["fileSynId"].values[0]
     clinical_ent = syn.get(clinical_link_synid, followLink=True)
     clinical_df = pandas.read_csv(clinical_ent.path, sep="\t", skiprows=4)
+    # get column list for the form
+    column_list = column_mapping_table.loc[
```

Comment:
The column mapping table is a fileview right?

Reply:
No, it is a table: https://www.synapse.org/Synapse:syn20945902/tables/